### PR TITLE
Enhance testing for pending request count

### PR DESCRIPTION
**Pull request by tanmayv25**
This tests the fix in PR https://github.com/triton-inference-server/core/pull/286

### CODE REVIEW

File: qa/L0_metrics/metrics_queue_size_test.py [hunk 1]
```diff
@@ -194,6 +221,13 @@ def test_dynamic_batch_scheduler(self):
         batch_size = self.max_batch_size
         self._test_helper(model_name, batch_size, self._send_async_requests)
 
+    def test_fail_max_queue_size(self):
+        model_name = "max_queue_size"
+        batch_size = self.max_batch_size
+        self._test_helper(
+            model_name, batch_size, self._send_async_requests, max_queue_size=4
+        )
```
Comment (rmccorm4) [inline qa/L0_metrics/metrics_queue_size_test.py:229]: Can you add a comment on test expectations? At a glance it looks like it wants to fill up the queue to max queue size and assert further requests are rejected/failed, but would be nice to have that summary associated with the test block.

Comment (rmccorm4) [reply qa/L0_metrics/metrics_queue_size_test.py:229]: LGTM otherwise, just want the description for future readers

Comment (tanmayv25) [reply qa/L0_metrics/metrics_queue_size_test.py]: Done!

**Review by rmccorm4 — APPROVED** [review]
LGTM, just double checking if the build.py version changes should be included when merging or cherry picking

**Comment by tanmayv25** [comment]
> LGTM, just double checking if the build.py version changes should be included when merging or cherry picking

Can you elaborate? There are no changes introduced in `build.py` by this PR.

**Pull request merged**